### PR TITLE
Generate Component/Service

### PR DESCRIPTION
**Pull request by ZimGil**
This pull request solves issue #5 
Added a "Generate" component that allows the user to select an Angular Class Type to create and name it.

### CODE REVIEW

File: src/app/generate/generate.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<h1>Generate</h1>
+<div>Name & Type</div>
+<input type="text" [(ngModel)]="name">
+<select (change)="selectAngularClassType($event.target.value)">
+    <option *ngFor="let type of optionalTypes" [value]="type.id">{{ type.displayName }}</option>
+</select>
+<button (click)="generateAngularClass()">Generate</button>
```
Comment (ortichon) [inline src/app/generate/generate.component.html]: please add an empty line at the end of the file, remember to ask me why :)

File: src/app/generate/generate.component.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  generateAngularClass(): void {
+    const request = {
+      name: this.name,
+      type: this.type
+    }
+    console.log('request', request);
+  }
+}
+
+export interface AngularClassOptions {
```
Comment (ortichon) [inline src/app/generate/generate.component.ts]: move to separate file

File: src/app/generate/generate.component.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  id: AngularClass;
+  displayName: string;
+}
+
+export enum AngularClass {
```
Comment (ortichon) [inline src/app/generate/generate.component.ts]: move to separate file

File: src/app/generate/generate.component.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  Component = 0,
+  Service = 1
+}
```
Comment (ortichon) [inline src/app/generate/generate.component.ts]: add empty line at the end of the file

**Comment by ZimGil** [comment]
Code adjusted to change request,
please review again, @ortichon

**Comment by ZimGil** [comment]
Code adjusted to change request,
please review again, @ortichon

File: src/app/app.component.html [hunk 5]
```diff
@@ -1,2 +1,3 @@
 <h1>Angular CLI to UI</h1>
-<app-serve></app-serve>
\ No newline at end of file
+<app-serve></app-serve>
+<app-generate></app-generate>
```
Comment (ortichon) [inline src/app/app.component.html]: Newline at EOF (end of file)

File: src/app/components/generate/generate.component.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import { AngularClassOptions } from '../../models/angular-class-options.interface';
+
+@Component({
+  selector: 'app-generate',
+  templateUrl: './generate.component.html',
+  styleUrls: ['./generate.component.css']
+})
+export class GenerateComponent {
+
+  name = "Name";
```
Comment (ortichon) [inline src/app/components/generate/generate.component.ts]: please use single quotes for strings in components

File: src/app/models/angular-class-options.interface.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import { AngularClass } from '../models/angular-class.enum';
+
+export interface AngularClassOptions {
+    id: AngularClass;
+    displayName: string;
+  }
```
Comment (ortichon) [inline src/app/models/angular-class-options.interface.ts]: EOF

File: src/app/models/angular-class.enum.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export enum AngularClass {
+    Component = 0,
+    Service = 1
+  }
+
```
Comment (ortichon) [inline src/app/models/angular-class.enum.ts]: it looks line EOF but you probably have a white-space there

File: src/app/components/generate/generate.component.html [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<h1>Generate</h1>
+<div>Name & Type</div>
+<input type="text" [(ngModel)]="name">
+<select [(ngModel)]="type">
+    <option [value]="none" selected>Choose a Class</option>
```
Comment (ortichon) [inline src/app/components/generate/generate.component.html]: I don't like this solution. I would like to have a default selection, should be the most used one (component?)

File: src/app/components/generate/generate.component.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  generateAngularClass(): void {
+    const request = {
+      name: this.name,
+      type: this.type
+    }
+    console.log('request', request);
+  }
+
+  isTypeValid(): boolean {
+    return this.type !>= 0;
```
Comment (ortichon) [inline src/app/components/generate/generate.component.ts]: I don't think we'll need this function anymore after the removal of non-selection options. 
but anyway this is much cleaner:
```typescript
isTypeValid(): boolean {
  return this.type < 0;
}
```

`x !>= y` is the same as `x < y`

File: src/app/components/generate/generate.component.ts [hunk 11]
```diff
@@ -10,8 +10,8 @@ import { AngularClassOptions } from '../../models/angular-class-options.interfac
 })
 export class GenerateComponent {
 
-  name = "Name";
-  type: AngularClass;
+  name = 'Name';
+  type = AngularClass.Component.toString();
```
Comment (ZimGil) [inline src/app/components/generate/generate.component.ts]: does that make sense?

**Comment by ZimGil** [comment]
Adjustments made,
component is the default selection now.

**Pull request merged**